pkg/asset/installconfig/gcp/validation.go

```diff
@@ -84,6 +91,20 @@ func ValidateInstanceType(client API, fieldPath *field.Path, project, zone, inst
 	return allErrs
 }
 
+func validateServiceAccountPresent(sess *Session, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.GCP.NetworkProjectID != "" {
+		if sess.Credentials != nil && sess.Credentials.JSON == nil {
+			if ic.ControlPlane.Platform.GCP != nil && ic.ControlPlane.Platform.GCP.ServiceAccount == "" {
+				errMsg := "service account must be provided when credentials are missing"
```

Comment:
nit: perhaps

`service account must be provided when authentication credentials do not provide a service account`